foundations/foundation-metrics/src/main/java/io/servicecomb/foundation/metrics/MetricsServoRegistry.java

```diff
@@ -0,0 +1,463 @@
+ */
+
+package io.servicecomb.foundation.metrics;
+
+import com.netflix.config.DynamicPropertyFactory;
+import com.netflix.hystrix.HystrixCommandMetrics;
+import com.netflix.hystrix.HystrixEventType;
+import com.netflix.servo.DefaultMonitorRegistry;
+import com.netflix.servo.annotations.DataSourceType;
+import com.netflix.servo.monitor.AbstractMonitor;
+import com.netflix.servo.monitor.BasicCompositeMonitor;
+import com.netflix.servo.monitor.Gauge;
+import com.netflix.servo.monitor.Informational;
+import com.netflix.servo.monitor.Monitor;
+import com.netflix.servo.monitor.MonitorConfig;
+import com.netflix.servo.publish.BasicMetricFilter;
+import com.netflix.servo.publish.CounterToRateMetricTransform;
+import com.netflix.servo.publish.FileMetricObserver;
+import com.netflix.servo.publish.MetricObserver;
+import com.netflix.servo.publish.MonitorRegistryMetricPoller;
+import com.netflix.servo.publish.PollRunnable;
+import com.netflix.servo.publish.PollScheduler;
+
+import io.servicecomb.foundation.metrics.performance.MetricsDataMonitor;
+import io.servicecomb.foundation.metrics.performance.QueueMetrics;
+
+import java.io.File;
+import java.lang.management.ManagementFactory;
+import java.lang.management.MemoryMXBean;
+import java.lang.management.MemoryUsage;
+import java.lang.management.OperatingSystemMXBean;
+import java.lang.management.ThreadMXBean;
+import java.lang.reflect.Field;
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.util.ReflectionUtils;
+
+import rx.functions.Func0;
+import rx.functions.Func2;
+
+/**
+ * Implementation of metrics preparation and servo registry.
+ */
+public class MetricsServoRegistry implements InitializingBean {
+
+  private static final String METRICS_POLL_TIME = "cse.metrics.polltime";
+
+  private static final String FILENAME = "cse.metrics.file.name";
+
+  private static final String FILEPATH = "cse.metrics.file.path";
+
+  private MetricsDataMonitor localMetrics = new MetricsDataMonitor();
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(MetricsServoRegistry.class);
+
+  /*
+   * Added getter for unit test of local metrics.
+   * 
+   * @return Local metric reference
+   */
+  public MetricsDataMonitor getLocalMetrics() {
+    return localMetrics;
+  }
+
+  /**
+   * Get the initial metrics and register with servo.
+   */
+  public void initMetricsPublishing() {
+
+    /* list of monitors */
+    List<Monitor<?>> monitors = getMetricsMonitors();
+
+    MonitorConfig commandMetricsConfig = MonitorConfig.builder("CSE_" + "test name").build();

```

Comment:
test name?

Reply:
this is fixed

Reply:
Please keep sending the changes in this PR.